spec/lib/saml/post_url_service_spec.rb

```diff
@@ -603,52 +603,105 @@
-            let(:expected_redirect_url_param) do
-              expected_login_redirect_url.to_query('redirect_url')
-            end
-            let(:expected_login_redirect_url) do
-              "#{values[:base_redirect]}#{SAML::URLService::LOGIN_REDIRECT_PARTIAL}?type=idme"
-            end
+            context 'with verified user' do
+              let(:user) { build(:user, :loa3) }
+              let(:expected_redirect_url_param) do
+                expected_login_redirect_url.to_query('redirect_url')
+              end
+              let(:expected_login_redirect_url) do
+                "#{values[:base_redirect]}#{SAML::URLService::LOGIN_REDIRECT_PARTIAL}?type=idme"
+              end
 
-            context 'when tracker application is within TERMS_OF_USE_ENABLED_CLIENTS' do
-              let(:application) { SAML::URLService::TERMS_OF_USE_ENABLED_CLIENTS.first }
+              context 'when tracker application is within TERMS_OF_USE_ENABLED_CLIENTS' do
+                let(:application) { SAML::URLService::TERMS_OF_USE_ENABLED_CLIENTS.first }
 
-              context 'and authentication is occuring on a review instance' do
-                let(:review_instance_slug) { 'some-review-instance-slug' }
-                let(:review_instance_url) { "#{review_instance_slug}.review.vetsgov-internal" }
+                context 'and authentication is occuring on a review instance' do
+                  let(:review_instance_slug) { 'some-review-instance-slug' }
+                  let(:review_instance_url) { "#{review_instance_slug}.review.vetsgov-internal" }
 
-                before { allow(Settings).to receive(:review_instance_slug).and_return(review_instance_slug) }
+                  before { allow(Settings).to receive(:review_instance_slug).and_return(review_instance_slug) }
 
-                it 'has a login redirect url as a parameter embedded in review instance terms of use page' do
-                  expect(subject.terms_of_use_redirect_url)
-                    .to eq("http://#{review_instance_url}/terms-of-use?#{expected_redirect_url_param}")
+                  it 'has a login redirect url as a parameter embedded in review instance terms of use page' do
+                    expect(subject.terms_of_use_redirect_url)
+                      .to eq("http://#{review_instance_url}/terms-of-use?#{expected_redirect_url_param}")
+                  end
+                end
+
+                context 'and authentication is not occurring on a review instance' do
+                  it 'has a login redirect url as a parameter embedded in terms of use page with success' do
+                    expect(subject.terms_of_use_redirect_url)
+                      .to eq("#{values[:base_redirect]}/terms-of-use?#{expected_redirect_url_param}")
+                  end
                 end
               end
 
-              context 'and authentication is not occurring on a review instance' do
+              context 'when tracker application is nil' do
```

Comment:
This is an issue with the previous specs, but `when tracker application is nil` actually ends up being the same flow as `when tracker application is within TERMS_OF_USE_ENABLED_CLIENTS`. This is because of this line in `post_url_service`:
`application = @tracker&.payload_attr(:application) || 'vaweb'`

Could you basically just copy the context from `when tracker application is within TERMS_OF_USE_ENABLED_CLIENTS` and put it inside the `when tracker application is nil` context? (I still think it's important to test both